apps/rewards/app/components/Panel/NewReward/NewReward.js

```diff
@@ -653,7 +743,10 @@ class NewRewardClass extends React.Component {
       dateStart,
       dateEnd,
       disbursements,
+      disbursementBlocks,
     } = this.state
+    this.props.app.web3Eth('getBlockNumber')
+      .subscribe(this.setDisbursementBlocks)
```

Comment:
`showSummary` is run during `render`. `render` should not have side effects, such as setting state.

Let's move this calculation to `componentDidMount` (if we were using a functional component, we could `useEffect`). Then we can fetch the value and set the state once, on initial load of the NewReward component, rather than on every render pass.

Reply:
It also doesn't looks like this is getting retriggered on each modification to the reward. For example if a user is bouncing back between the configuration and confirmation panes, the block numbers, and number of blocks needed to be calculated should be updating. It doesn't look like this is happening: 
![image](https://user-images.githubusercontent.com/17910833/68342743-dc63e600-00b0-11ea-934f-8cde93456032.png)

~5700 blocks between a weekly reward is not correct, and an additional block was not calculated for the additional disbursement I generated upon revision

Reply:
These two issues are indeed related and by solving one I solved the other. Instead of putting `setDisbursementBlocks` in `render`, I put it in `onSubmitDraft`, where it gets executed right when appropriate. Thanks, guys, for these very appropriate and illustrative remarks!